### PR TITLE
Fix [Feature Request] Dropdown in ButtonGroup(#12815)

### PR DESCRIPTION
**Pull request by bluejfox**
Please make sure these boxes are checked before submitting your PR, thank you!

* [x] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md) | [Español](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.es.md)).
* [x] Make sure you are merging your commits to `dev` branch.
* [x] Add some descriptions and refer relative issues for you PR.

Fix [Feature Request] Dropdown in ButtonGroup(#12815)
Now ButtonGroup support Dropdown which be place as ButtonGroup's last-child.

### CODE REVIEW

**Comment by bluejfox** [comment]
For implement below scenario.

image

[Demo Link](https://jsfiddle.net/phoenixlhan/g56abo3L/)

File: packages/theme-chalk/src/button.scss [hunk 1]
```diff
@@ -186,7 +186,7 @@
   display: inline-block;
   vertical-align: middle;
 
-  & .el-button {
+  & > .el-button {
```
Comment (wacky6) [inline packages/theme-chalk/src/button.scss]: why is this change necessary?

File: packages/theme-chalk/src/button.scss [hunk 2]
```diff
@@ -232,6 +232,19 @@
       z-index: 1;
     }
   }
+  
+  & > .el-dropdown {
+    & > .el-button {
+      border-top-left-radius: 0;
+      border-bottom-left-radius: 0;
+      
+      @each $type in (primary, success, warning, danger, info) {
```
Comment (ziyoung) [inline packages/theme-chalk/src/button.scss]: Is it necessary to use a loop?
```css
border-left-color: rgba($--color-white, 0.5);
```
This also should work.

Comment (bluejfox) [reply packages/theme-chalk/src/button.scss]: When Dropdown is the last child in ButtonGroup, the Button which in Dropdown isn't be considered the last child in the ButtonGroup, and the Dropdown is usually used as the last child in the ButtonGroup, so I add below loop style.
```css
@each $type in (primary, success, warning, danger, info) {
  &.el-button--#{$type} {
    border-left-color: rgba($--color-white, 0.5);
  }
}
```
image

Comment (bluejfox) [reply packages/theme-chalk/src/button.scss]: @ziyoung Sorry, I misunderstood your point.I had modified style source.Pls confirm it.
